core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java

```diff
@@ -144,19 +174,50 @@ private static void putReportedMetrics(IndexedSet<Metric> metricSet, String inst
       // If a metric is COUNTER, the value sent via RPC should be the incremental value; i.e.
       // the amount the value has changed since the last RPC. The master should equivalently
       // increment its value based on the received metric rather than replacing it.
-      if (!metricSet.add(metric)) {
-        Metric oldMetric = metricSet.getFirstByField(FULL_NAME_INDEX, metric.getFullMetricName());
-        if (metric.getMetricType() == MetricType.COUNTER) {
-          if (metric.getValue() != 0L) {
-            oldMetric.addValue(metric.getValue());
-          }
-        } else {
-          oldMetric.setValue(metric.getValue());
+      if (metric.getMetricType() == MetricType.COUNTER) {
+        String metricKeyName = metric.getInstanceType() + "." + metric.getName();
```

Comment:
The `metricKeyName` is used for matching the `MetricKey.WORKER_BYTES_READ_UFS.getName()`.

In the `MetricKey`, I gave the prefix `MASTER_/WORKER_/CLIENT_<ACTUAL_METRIC_NAME>` because the `<ACTUAL_METRIC_NAME>` may be duplicated in worker or client metrics.

Yeah this may be expensive, would you think it will be better to use a struct directly 
```
ConcurrentHashMap<ClusterCounterKey, Counter> mClusterCounters
struct ClusterCounterkey{
  String instanceType
  String metricName
}
```
but the string concatenation is still needed when comparing with the `MetricKey.WORKER_BYTES_READ_UFS.getName()` and `MetricKey.WORKER_BYTES_WRITTEN_UFS.getName()`.

Reply:
I remember string concat being quite expensive. One alternative is check instanceType == WORKER and metricName.equals(BYTES_READ_UFS)?

Reply:
Remove the potentially expensive string concatenation